### PR TITLE
[flash_ctrl/dv] Fix sec_cm assertions

### PR DESCRIPTION
**Pull request by nasahlpa**
This commit disables assertions that get triggered during sec_cm assertions. Disabling these assertions is valid
as, on a FI, the `u_sramreqfifo`, `u_rspfifo` outputs could be manipulated. Although this error is captured by `intg_error_o` the current flash DV env is not aware of this and triggers invalid data assertions.

Moreover, we need to turn off the flash_access_after_disabled function when inducing faults into the counters of the read FIFOS. The reason for that is:
- A fault in these FIFOs could manipulate a TL-UL message
- This is detected by the DV environment by sensing the fatal alert
- If we would now call the flash_access_after_disabled function while there still is a manipulated TL-UL message in the system we could get a timeout
- Instead, we shouldn't rely that we still can access the flash after disable when there is a fault injection attack going on

Partly addresses #23567, as the `sram_ctrl` DV environment needs to be adapted similarly.

### CODE REVIEW

**Comment by nasahlpa** [comment]
Opening this PR as pass rate for the previously failing `sec_cm` tests look good:
<details>
<!DOCTYPE html>
<html lang="en">
<body>
<div class="results" style='font-family:"Trebuchet MS", Arial, Helvetica, sans-serif; margin:0 auto; max-width:960px; padding-left:40px; padding-right:40px; padding-top:15px; position:relative; width:80%' width="80%">
<h2 style="padding-bottom:15px; text-align:center" align="center">FLASH_CTRL Simulation Results</h2>
<h3 style="padding-bottom:15px; text-align:center" align="center">Tuesday June 11 2024 07:30:59 UTC</h3>
<h3 style="padding-bottom:15px; text-align:center" align="center">GitHub Revision: <a href="https://github.com/lowrisc/opentitan/tree/2546d4be6e03b4785fd28de9a3e21812d0da2b80"><code>2546d4be6e</code></a></h3>
<h3 style="padding-bottom:15px; text-align:center" align="center">Branch: tlul_adapter_sec_cm</h3>
<h3 style="padding-bottom:15px; text-align:center" align="center"><a href="https://opentitan.org/book/hw/top_earlgrey/ip_autogen/flash_ctrl/data/flash_ctrl_testplan.html">Testplan</a></h3>
<h3 style="padding-bottom:15px; text-align:center" align="center">Simulator: VCS</h3>
<h3 style="padding-bottom:15px; text-align:center" align="center">Test Results</h3>
<table style="border:1px solid #f2f2f2; border-collapse:collapse; display:table; margin:2% auto; table-layout:auto; text-align:center; vertical-align:middle; width:90%" align="center" valign="middle" width="90%">
<thead>
<tr>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Stage</th>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Name</th>
<th align="left" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Tests</th>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Max Job Runtime</th>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Simulated Time</th>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Passing</th>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Total</th>
<th align="center" style="background-color:#3D1067; color:white; padding-bottom:12px; padding-top:12px; text-transform:uppercase; border:1px solid #f2f2f2; padding:8px" bgcolor="#3D1067">Pass Rate</th>
</tr>
</thead>
<tbody>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">tl_intg_err</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_ctrl_fsm_sparse</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_phy_fsm_sparse</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_phy_prog_fsm_sparse</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_ctr_redun</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_fifo_ctr_redun</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_mem_tl_lc_gate_fsm_sparse</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">sec_cm_prog_tl_lc_gate_fsm_sparse</td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px">flash_ctrl_sec_cm</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">1.460h</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">24.031ms</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">V2S</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px"><strong>TOTAL</strong></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
<tr>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="left" style="border:1px solid #f2f2f2; padding:8px"><strong>TOTAL</strong></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px"></td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td align="center" style="border:1px solid #f2f2f2; padding:8px">5</td>
<td class="c10" align="center" style="background-color:#57EF57; color:#000; border:1px solid #f2f2f2; padding:8px" bgcolor="#57EF57">100.00</td>
</tr>
</tbody>
</table>
</div>
</body>
</html>
</details>

File: hw/ip_templates/flash_ctrl/dv/env/seq_lib/flash_ctrl_common_vseq.sv [hunk 1]
```diff
@@ -94,6 +94,8 @@ class flash_ctrl_common_vseq extends flash_ctrl_otf_base_vseq;
       end
       sec_cm_fi_ctrl_svas(if_proxy, .enable(1));
       // issue hard reset for fatal alert to recover
+      `uvm_info(`gfn, $sformatf("issue a reset"), UVM_LOW)
+      `uvm_info(`gfn, $sformatf("issue a reset"), UVM_HIGH)
```
Comment (vogelpi) [inline hw/ip_templates/flash_ctrl/dv/env/seq_lib/flash_ctrl_common_vseq.sv:98]: Is this change intentional?

**Review by vogelpi — APPROVED** [review]
This looks good to me, thanks @nasahlpa for your work on this. As discussed offline with @GregAC and @rswarbrick we believe this should be handled in DV as done with this PR.

@matutem , I can give you a quick update on this later today.

**Comment by nasahlpa** [comment]
Closing this PR as it is superseded by #23636.
